components/view_user_group_modal/view_user_group_list_item/view_user_group_list_item.tsx

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+
+import {Group} from 'mattermost-redux/types/groups';
+import {UserProfile} from 'mattermost-redux/types/users';
+
+import Avatar from 'components/widgets/users/avatar';
+import * as Utils from 'utils/utils.jsx';
+import LocalizedIcon from 'components/localized_icon';
+import {t} from 'utils/i18n';
+import {ActionResult} from 'mattermost-redux/types/actions';
+
+export type Props = {
+    groupId: string;
+    user: UserProfile;
+    group: Group;
+    decrementMemberCount: () => void;
+    permissionToLeaveGroup: boolean;
+    actions: {
+        removeUsersFromGroup: (groupId: string, userIds: string[]) => Promise<ActionResult>;
+    };
+}
+
+const ViewUserGroupListItem = (props: Props) => {
+    const {
+        user,
+        group,
+    } = props;
+
+    const removeUserFromGroup = async (userId: string) => {
```

Comment:
You can use `useCallback` here and create a function with 0 parameters and use the `user.id` from the context, that way you can pass that memoized callback to the button component avoiding re-rendering it on every render.